db/item/memo/addr_post.go

```diff
@@ -48,21 +48,29 @@ func (p *AddrPost) Serialize() []byte {
 
 func (p *AddrPost) Deserialize([]byte) {}
 
-func GetAddrPosts(ctx context.Context, addrs [][25]byte, newest bool) ([]*AddrPost, error) {
+func GetAddrPosts(ctx context.Context, addrs [][25]byte, newest bool, start time.Time) ([]*AddrPost, error) {
 	var shardPrefixes = make(map[uint32][][]byte)
 	for i := range addrs {
 		shard := client.GetByteShard32(addrs[i][:])
 		shardPrefixes[shard] = append(shardPrefixes[shard], addrs[i][:])
 	}
 	shardConfigs := config.GetQueueShards()
 	var addrPosts []*AddrPost
+	var i = 0
 	for shard, prefixes := range shardPrefixes {
+		var startByte []byte
+		if !jutil.IsTimeZero(start) {
+			startByte = jutil.CombineBytes(addrs[i][:], jutil.GetTimeByteNanoBig(start))
```

Comment:
What if the 1st address is in the 2nd shard? What if there are more addresses than shards?